packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -308,97 +189,201 @@ class CodeSizeBodyState extends State<CodeSizeBody>
+  Row _buildDiffViewControls() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
       children: [
-        // const Text('Imported file:'),
-        // const SizedBox(height: 8.0),
-        Text(
-          importedFile,
-          style: const TextStyle(color: Colors.white54),
-          textAlign: TextAlign.center,
+        _buildDiffTreeTypeDropdown(),
+        const SizedBox(width: defaultSpacing),
+        clearButton(
+          busy: diffRoot == null,
+          onPressed: controller.clearDiff,
         ),
       ],
     );
   }
 
-  Column buildImportButton() {
+  DropdownButtonHideUnderline _buildDiffTreeTypeDropdown() {
+    return DropdownButtonHideUnderline(
+      child: DropdownButton<DiffTreeType>(
+        value: controller.activeDiffTreeType.value,
+        items: [
+          _buildMenuItem(DiffTreeType.combined),
+          _buildMenuItem(DiffTreeType.increaseOnly),
+          _buildMenuItem(DiffTreeType.decreaseOnly),
+        ],
+        onChanged: controller.diffRoot.value == null
+            ? null
+            : (newDiffTreeType) {
+                controller.changeActiveDiffTreeType(newDiffTreeType);
+              },
+      ),
+    );
+  }
+
+  DropdownMenuItem _buildMenuItem(DiffTreeType diffTreeType) {
+    return DropdownMenuItem<DiffTreeType>(
+      value: diffTreeType,
+      child: Text(diffTreeType.display),
+    );
+  }
+
+  Widget _buildBody() {
+    if (diffRoot != null) {
+      return _buildTreemapAndTableSplitView();
+    } else {
+      return _buildImportDiffView();
+    }
+  }
+
+  Widget _buildTreemapAndTableSplitView() {
+    return Split(
+      axis: Axis.vertical,
+      children: [
+        _buildTreemap(),
+        CodeSizeDiffTable(rootNode: diffRoot),
+      ],
+      initialFractions: const [
+        initialFractionForTreemap,
+        initialFractionForTreeTable,
+      ],
+    );
+  }
+
+  Widget _buildImportDiffView() {
+    return Padding(
+      padding: const EdgeInsets.all(defaultSpacing),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          Expanded(
+            child: DualFileImportContainer(
+              actionText: 'Analyze Diff',
+              onAction: controller.loadFakeDiffTree,
+            ),
+          ),
+          const SizedBox(height: defaultSpacing),
+          _buildHelpText(),
+        ],
+      ),
+    );
+  }
+
+  Column _buildHelpText() {
     return Column(
-      mainAxisAlignment: MainAxisAlignment.center,
       children: [
-        const Icon(Icons.file_upload),
-        RaisedButton(
-          // TODO(peterdjlee): Prompt file picker to choose a snapshot file.
-          onPressed: onImport,
-          child: const Text('Import file'),
+        Text(
+          'We currently only support instruction sizes and v8 snapshot profile outputs.',
+          style:
+              TextStyle(color: Theme.of(context).colorScheme.chartAccentColor),
         ),
       ],
     );
   }
```

Comment:
this method is duplicated. Make it a top-level private method at the bottom of the file